src/components/App/App.js

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import styled from 'styled-components/macro'
+import { Route, Switch } from 'react-router-dom'
+import WorkoutPage from '../WorkoutPage'
+import MusicPage from '../MusicPage'
+import PlaylistPage from '../PlaylistPage'
+import data from '../tracksShort.json'
+import { rangeMin, rangeMax } from '../../services/cadenceRange'
+import convertCadenceTempo from '../../services/convertCadenceTempo'
+import roundedTempo from '../../services/roundedTempo'
+import { results } from '../staticWorkoutList.json'
+
+export default function App() {
+  const { push } = useHistory()
+
+  const [warmupSongs, setWarmupSongs] = useState([])
+  const [intervalsTSongs, setIntervalsTSongs] = useState([])
+  const [cooldownSongs, setCooldownSongs] = useState([])
+  const [playlist, setPlaylist] = useState([])
+
+  return (
+    <AppLayout>
+      <Switch>
+        <Route exact path="/">
+          <WorkoutPage results={results} />
+        </Route>
+        <Route
+          path="/music"
+          render={props => (
+            <MusicPage {...props} onCreatePlaylist={createPlaylist} />
+          )}
+        />
+
+        <Route path="/playlist">
+          <PlaylistPage
+            playlist={playlist}
+            warmupSongs={warmupSongs}
+            intervalsTSongs={intervalsTSongs}
+            cooldownSongs={cooldownSongs}
+          />
+        </Route>
+      </Switch>
+    </AppLayout>
+  )
+
+  function createPlaylist(values, workout) {
```

Comment:
Well done! 👍 🙂 
A thought: this functionality is pretty complex and it's a lot of repetition. If you do a refactor of this code, you could think about having more service functions so you can reduce the repetition a bit. We can also discuss this next week if you have questions about that.